### ISSUE TITLE
 [Announcement] Please stay tuned with this thread.  

### ISSUE THREAD

**Issue by eastskykang** [cmm-21/a5#4]
There're some minor updates in README.md file. 

They are not very relevant to your tasks but rather some minor changes or typo fixes. It wouldn't matter if you don't merge these update to your repo. 

**Update**: now there's an important update in main branch. Please see this: https://github.com/cmm-21/a5/issues/4#issuecomment-831804671

**Update2**: please do not add comment here rather create a new thread if you have any question.

**Comment by eastskykang** [comment]
We are going to change the task of Ex.4 a bit. I will update code and READEME.md as soon as it's ready. More details and instructions will be announced in next tutorial session.

Please see this #1 thread for related discussions.

**Comment by breadli428** [comment]
Hi,

Where can we see and how can we merge the updates?

Edit: Well I saw the change in this repo. Is there any suggestion to Git beginner to merge the change in this repo to my private one?

Thanks!

**Comment by eastskykang** [comment]
### Update on skeleton code and Ex.4

Hi all, 

Ex.4 and some minor code updates are pushed to main branch. 

(I strongly recommend you to back up your code to another directory first) Please merge the changes to your branch by.

```sh
# in project directory
$ git remote add origin-start-code https://github.com/cmm-21/a5.git
$ git pull origin-start-code main --rebase
```

and fix conflicts if there's any.

**Comment by eastskykang** [comment]
### Ex.3

Concerning Ex. 3, you are okay to change the integration method. I was misleading in the last tutorial session (April 29). What I wanted to say was, you don't need any other damping method.

**Comment by eastskykang** [comment]
### Demo videos

Please find the demo videos:

- baselines: https://youtu.be/aOyybxvSG4A
- Ex.4: https://youtu.be/v1zMGm-7id4

**Comment by eastskykang** [comment]
### Ex.4

You have to reproduce exactly same behavior presented in the video above, to get a full points for Ex.4

**Comment by nicolas-git** [comment]
It is normal that ex.4 won't work correctly at all times if we apply a force to it which initiates angular velocities, right?

**Comment by iqiac** [comment]
![image](https://user-images.githubusercontent.com/58211069/117533975-c7427d00-afef-11eb-9029-a2ee5582f853.png)
Hi, while trying to pull from github for task 4 I'm getting this error. Looks like a typo maybe.. How do I correct this?

Edit: nvm, seems to be a typo from my side
I did `git remote remove origin-start-code` and the did the same again without typo and it worked

**Comment by eastskykang** [comment]
@nicolas-git 

yes, we are not going to test collision simulation with external force or torque (+ w/ angular velocity). Please just make sure if it works as the demo video describes.

**Comment by eastskykang** [comment]
@nicolas-git and make sure your implementation follows the step-by-step instructions. If there's some error, there might be some deduction on your grade. 

p.s. Something could be wrong if it completely goes to nuts. (our model is imperfect, but reasonably good to simulate a collision between ball and ground.) I can imagine there could be a common mistake that is giving a wrong value for **r**<sub>a</sub> as well as d**r**<sub>a</sub>/dt. Remember, the bottom of the ball is not a fixed point. It changes as the ball rotates. (I quickly tested with my implementation and it looks fine with non-zero angular velocity)

**Comment by eastskykang** [comment]
@nicolas-git 

This is what I got from my implementation: https://youtu.be/6RTHyJI2zUc

If you have a further question, please add your comment on #18

**Comment by eastskykang** [comment]
======= PLEASE DO NOT ADD COMMENT BELOW THIS LINE =======

**Comment by eastskykang** [comment]
### Ex.4 (update)

I admit my instruction was a bit vague. I will give you full point if your implementation reproduce exactly same result with the demo I posted. You don't need to worry about scenario with an external force and angular velocity.

But, please try to modify your code to make it work with external force (and torque) and angular velocity > 0 if you have some time (and motivation). It does not require much effort.
